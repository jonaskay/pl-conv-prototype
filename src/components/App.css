.App {
  box-sizing: border-box;
  display: flex;
  height: 100vh; 
  margin: auto;
  padding: 1.5rem;
  width: 75rem;
}


.App:before {
  content: '';
  height: 7rem;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 0;
}


.App-col1 {
  width: 50rem;
}

.App-col2 {
  box-sizing: border-box;
  padding-left: 1.5rem;
  width: 22rem;
}
