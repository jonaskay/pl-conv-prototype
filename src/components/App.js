import React, { Component } from 'react';
import './App.css';
import Draft from './Draft';
import ConvertMenu from './ConvertMenu';
import ConvertButton from './ConvertButton';
import Modal from './Modal';
import FileForm from './FileForm';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {fileData: null, playlistName: ''};
  }

  handleFileSubmit = (e) => {
    e.preventDefault();
    this.setState({fileData: this.props.demoData, playlistName: this.props.demoName});
  }
  
  handleNameChange = (e) => {
    this.setState({playlistName: e.target.value});
  }

  render() {
    return (
      <div className="App">
        {!this.state.fileData && 
          <Modal>
            <FileForm fileName={this.props.demoName} onSubmit={this.handleFileSubmit} />
          </Modal>
        }
        <div className="App-col1">
          <Draft 
            items={this.state.fileData} 
            name={this.state.playlistName} 
            onNameChange={this.handleNameChange} />
        </div>
        <div className="App-col2">
          <ConvertMenu />
          <ConvertButton />
        </div>
      </div>
    );
  }
}

export default App;
