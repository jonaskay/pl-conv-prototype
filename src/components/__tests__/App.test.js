import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';
import jsonData from '../../tracks.json';

Enzyme.configure({ adapter: new Adapter() });

it('renders the app', () => {
  const component = shallow(<App demoName="foobar" demoData={jsonData.slice(0, 1)} />);
  expect(component).toMatchSnapshot();
});

it('renders file data after file form submit', () => {
  const component = mount(<App demoName="foobar"  demoData={jsonData.slice(0, 1)} />);
  component.find('.FileForm-form').simulate('submit');
  
  expect(component).toMatchSnapshot();
})

it('updates playlist name after name change', () => {
  const component = mount(<App />);
  component.setState({fileData: jsonData.slice(0, 1), fileName: 'Foobar'});
  component.find('.DraftNameInput').simulate('change', { target: { value: 'Baz' } });

  expect(component).toMatchSnapshot();
});

