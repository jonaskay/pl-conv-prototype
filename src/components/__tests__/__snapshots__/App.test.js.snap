// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders file data after file form submit 1`] = `
<App
  demoData={
    Array [
      Object {
        "artist": "Cardi B",
        "id": 1,
        "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
        "name": "Bodak Yellow",
        "spotify_id": "2771LMNxwf62FTAdpJMQfM",
        "youtube_id": "PEGccV-NOm8",
      },
    ]
  }
  demoName="foobar"
>
  <div
    className="App"
  >
    <div
      className="App-col1"
    >
      <Draft
        items={
          Array [
            Object {
              "artist": "Cardi B",
              "id": 1,
              "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
              "name": "Bodak Yellow",
              "spotify_id": "2771LMNxwf62FTAdpJMQfM",
              "youtube_id": "PEGccV-NOm8",
            },
          ]
        }
        name="foobar"
        onNameChange={[Function]}
      >
        <div
          className="Draft"
        >
          <div
            className="Draft-col1"
          >
            <div
              className="Draft-header"
            >
              <div
                style={
                  Object {
                    "color": "rgba(255,255,255,.5)",
                  }
                }
              >
                <span>
                  foobar
                </span>
                 ▪ 
                <DraftItemCounter
                  items={
                    Array [
                      Object {
                        "artist": "Cardi B",
                        "id": 1,
                        "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
                        "name": "Bodak Yellow",
                        "spotify_id": "2771LMNxwf62FTAdpJMQfM",
                        "youtube_id": "PEGccV-NOm8",
                      },
                    ]
                  }
                >
                  <span>
                    1
                     
                    track
                  </span>
                </DraftItemCounter>
              </div>
              <DraftNameInput
                name="foobar"
                onChange={[Function]}
              >
                <input
                  className="DraftNameInput"
                  onChange={[Function]}
                  type="text"
                  value="foobar"
                />
              </DraftNameInput>
            </div>
            <div
              className="Draft-table"
            >
              <div
                className="Draft-table-header"
              >
                <div
                  style={
                    Object {
                      "flex": "none",
                      "width": "13.5rem",
                    }
                  }
                >
                  Name
                </div>
                <div
                  style={
                    Object {
                      "flex": "auto",
                    }
                  }
                >
                  Artist
                </div>
              </div>
              <div
                className="Draft-table-body"
              >
                <DraftItem
                  item={
                    Object {
                      "artist": "Cardi B",
                      "id": 1,
                      "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
                      "name": "Bodak Yellow",
                      "spotify_id": "2771LMNxwf62FTAdpJMQfM",
                      "youtube_id": "PEGccV-NOm8",
                    }
                  }
                  key="1"
                  onClick={[Function]}
                  selected={false}
                >
                  <div
                    className="DraftItem"
                    onClick={[Function]}
                  >
                    <div
                      style={
                        Object {
                          "flex": "none",
                          "width": "13.5rem",
                        }
                      }
                    >
                      Bodak Yellow
                    </div>
                    <div>
                      Cardi B
                    </div>
                  </div>
                </DraftItem>
              </div>
            </div>
          </div>
          <div
            className="Draft-col2"
          />
        </div>
      </Draft>
    </div>
    <div
      className="App-col2"
    >
      <ConvertMenu>
        <div
          className="ConvertMenu"
        >
          <div
            className="ConvertMenu-header"
          >
            Settings
          </div>
          <div
            className="ConvertMenu-body"
          >
            <div
              style={
                Object {
                  "color": "rgba(255,255,255,.7)",
                }
              }
            >
              Convert playlist to:
            </div>
            <ConvertMenuField
              name="Spotify"
            >
              <div
                className="ConvertMenuField"
              >
                <div>
                  <input
                    disabled={true}
                    type="checkbox"
                  />
                  <label
                    style={
                      Object {
                        "color": "rgba(255,255,255,.5)",
                        "marginLeft": ".5rem",
                      }
                    }
                  >
                    Spotify
                  </label>
                </div>
                <div>
                  <button
                    className="ConvertMenuField-button"
                  >
                    Connect with 
                    Spotify
                  </button>
                </div>
              </div>
            </ConvertMenuField>
            <ConvertMenuField
              name="YouTube"
            >
              <div
                className="ConvertMenuField"
              >
                <div>
                  <input
                    disabled={true}
                    type="checkbox"
                  />
                  <label
                    style={
                      Object {
                        "color": "rgba(255,255,255,.5)",
                        "marginLeft": ".5rem",
                      }
                    }
                  >
                    YouTube
                  </label>
                </div>
                <div>
                  <button
                    className="ConvertMenuField-button"
                  >
                    Connect with 
                    YouTube
                  </button>
                </div>
              </div>
            </ConvertMenuField>
          </div>
        </div>
      </ConvertMenu>
      <ConvertButton>
        <button
          className="ConvertButton"
        >
          Convert Playlist
        </button>
      </ConvertButton>
    </div>
  </div>
</App>
`;

exports[`renders the app 1`] = `
<div
  className="App"
>
  <Modal>
    <FileForm
      fileName="foobar"
      onSubmit={[Function]}
    />
  </Modal>
  <div
    className="App-col1"
  >
    <Draft
      items={null}
      name=""
      onNameChange={[Function]}
    />
  </div>
  <div
    className="App-col2"
  >
    <ConvertMenu />
    <ConvertButton />
  </div>
</div>
`;

exports[`updates playlist name after name change 1`] = `
<App>
  <div
    className="App"
  >
    <div
      className="App-col1"
    >
      <Draft
        items={
          Array [
            Object {
              "artist": "Cardi B",
              "id": 1,
              "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
              "name": "Bodak Yellow",
              "spotify_id": "2771LMNxwf62FTAdpJMQfM",
              "youtube_id": "PEGccV-NOm8",
            },
          ]
        }
        name="Baz"
        onNameChange={[Function]}
      >
        <div
          className="Draft"
        >
          <div
            className="Draft-col1"
          >
            <div
              className="Draft-header"
            >
              <div
                style={
                  Object {
                    "color": "rgba(255,255,255,.5)",
                  }
                }
              >
                <span>
                  Baz
                </span>
                 ▪ 
                <DraftItemCounter
                  items={
                    Array [
                      Object {
                        "artist": "Cardi B",
                        "id": 1,
                        "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
                        "name": "Bodak Yellow",
                        "spotify_id": "2771LMNxwf62FTAdpJMQfM",
                        "youtube_id": "PEGccV-NOm8",
                      },
                    ]
                  }
                >
                  <span>
                    1
                     
                    track
                  </span>
                </DraftItemCounter>
              </div>
              <DraftNameInput
                name="Baz"
                onChange={[Function]}
              >
                <input
                  className="DraftNameInput"
                  onChange={[Function]}
                  type="text"
                  value="Baz"
                />
              </DraftNameInput>
            </div>
            <div
              className="Draft-table"
            >
              <div
                className="Draft-table-header"
              >
                <div
                  style={
                    Object {
                      "flex": "none",
                      "width": "13.5rem",
                    }
                  }
                >
                  Name
                </div>
                <div
                  style={
                    Object {
                      "flex": "auto",
                    }
                  }
                >
                  Artist
                </div>
              </div>
              <div
                className="Draft-table-body"
              >
                <DraftItem
                  item={
                    Object {
                      "artist": "Cardi B",
                      "id": 1,
                      "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
                      "name": "Bodak Yellow",
                      "spotify_id": "2771LMNxwf62FTAdpJMQfM",
                      "youtube_id": "PEGccV-NOm8",
                    }
                  }
                  key="1"
                  onClick={[Function]}
                  selected={false}
                >
                  <div
                    className="DraftItem"
                    onClick={[Function]}
                  >
                    <div
                      style={
                        Object {
                          "flex": "none",
                          "width": "13.5rem",
                        }
                      }
                    >
                      Bodak Yellow
                    </div>
                    <div>
                      Cardi B
                    </div>
                  </div>
                </DraftItem>
              </div>
            </div>
          </div>
          <div
            className="Draft-col2"
          />
        </div>
      </Draft>
    </div>
    <div
      className="App-col2"
    >
      <ConvertMenu>
        <div
          className="ConvertMenu"
        >
          <div
            className="ConvertMenu-header"
          >
            Settings
          </div>
          <div
            className="ConvertMenu-body"
          >
            <div
              style={
                Object {
                  "color": "rgba(255,255,255,.7)",
                }
              }
            >
              Convert playlist to:
            </div>
            <ConvertMenuField
              name="Spotify"
            >
              <div
                className="ConvertMenuField"
              >
                <div>
                  <input
                    disabled={true}
                    type="checkbox"
                  />
                  <label
                    style={
                      Object {
                        "color": "rgba(255,255,255,.5)",
                        "marginLeft": ".5rem",
                      }
                    }
                  >
                    Spotify
                  </label>
                </div>
                <div>
                  <button
                    className="ConvertMenuField-button"
                  >
                    Connect with 
                    Spotify
                  </button>
                </div>
              </div>
            </ConvertMenuField>
            <ConvertMenuField
              name="YouTube"
            >
              <div
                className="ConvertMenuField"
              >
                <div>
                  <input
                    disabled={true}
                    type="checkbox"
                  />
                  <label
                    style={
                      Object {
                        "color": "rgba(255,255,255,.5)",
                        "marginLeft": ".5rem",
                      }
                    }
                  >
                    YouTube
                  </label>
                </div>
                <div>
                  <button
                    className="ConvertMenuField-button"
                  >
                    Connect with 
                    YouTube
                  </button>
                </div>
              </div>
            </ConvertMenuField>
          </div>
        </div>
      </ConvertMenu>
      <ConvertButton>
        <button
          className="ConvertButton"
        >
          Convert Playlist
        </button>
      </ConvertButton>
    </div>
  </div>
</App>
`;
